load() not working on brand new checkpoint

Hitting a Theano assertion. Anyone seen this before?

```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-6-f65624dca9da> in <module>()
      2 for c in checkpoints:
      3     with open(c, 'rb') as f:
----> 4         loops.append(load(f))

/u/wardefar/src/blocks/blocks/serialization.py in load(file_, name, use_cpickle)
    259         if '_parameters' in tar_file.getnames():
    260             p.persistent_load = _PersistentLoad(tar_file)
--> 261         return p.load()
    262 
    263 

/u/wardefar/src/theano/theano/compile/function_module.py in _constructor_Function(maker, input_storage, inputs_data)
   1026         assert (container.data is x) or \
   1027             (isinstance(x, numpy.ndarray) and (container.data == x).all()) or \
-> 1028             (container.data == x)
   1029     return f
   1030 

AssertionError: 
```

--- a/blocks/serialization.py
+++ b/blocks/serialization.py
@@ -563,10 +563,16 @@ class _PersistentLoad(object):
         if '_parameters' in tar_file.getnames():
             self.parameters = numpy.load(
                 tar_file.extractfile(tar_file.getmember('_parameters')))
+        self._cache = {}
 
     def __call__(self, id_):
-        components = _unmangle_parameter_name(id_)
-        return components[0](self.parameters[components[1]])
+        # As we empirically found out, this method can be called multiple
+        # times  with the same id_. That's why we need a cache here to
+        # avoid creating the same object more than once.
+        if not id_ in self._cache:
+            components = _unmangle_parameter_name(id_)
+            self._cache[id_] = components[0](self.parameters[components[1]])
+        return self._cache[id_]
 
 
 def _mangle_parameter_name(type_, name):

--- a/blocks/serialization.py
+++ b/blocks/serialization.py
@@ -569,7 +569,7 @@ class _PersistentLoad(object):
         # As we empirically found out, this method can be called multiple
         # times  with the same id_. That's why we need a cache here to
         # avoid creating the same object more than once.
-        if not id_ in self._cache:
+        if id_ not in self._cache:
             components = _unmangle_parameter_name(id_)
             self._cache[id_] = components[0](self.parameters[components[1]])
         return self._cache[id_]

